### ISSUE TITLE
use crf loss is zero

### ISSUE THREAD

**Issue by EricAugust** [bamtercelboo/pytorch_NER_BiLSTM_CNN_CRF#6]

**Comment by bamtercelboo** [comment]
Running the default program? Changed data?

**Comment by EricAugust** [comment]
changed data,  BIO format data.

**Comment by bamtercelboo** [comment]
@EricAugust 
Hi, Please show me the detailed model section profile information( [Model] in config.cfg file).

**Comment by EricAugust** [comment]
` 
[Embed]
pretrained_embed = False
zeros = False
avg = False
uniform = False
nnembed = True
pretrained_embed_file = 

[Data]
train_file = ./data//train.txt
dev_file = ./data/valid.txt
test_file = ./data/test.txt
max_count = -1
min_freq = 1
shuffle = True
epochs_shuffle = True

[Save]
save_pkl = True
pkl_directory = ./Save_pkl
pkl_data = pkl_data.pkl
pkl_alphabet = pkl_alphabet.pkl
pkl_iter = pkl_iter.pkl
pkl_embed = pkl_embed.pkl
save_dict = True
dict_directory = ./Save_dictionary
word_dict = dictionary_word.txt
label_dict = dictionary_label.txt
save_direction = ./Save_model
save_best_model_dir = ./Save_BModel
save_model = True
save_all_model = False
save_best_model = True
model_name = ner_model
rm_model = True

[Model]
average_batch = True
use_crf = True
use_char = True
model_bilstm = True
model_bilstm_context = False
lstm_layers = 1
embed_dim = 300
embed_finetune = True
lstm_hiddens = 200
dropout_emb = 0.5
dropout = 0.5
max_char_len = 20
char_dim = 200
conv_filter_sizes = 3,4,5
conv_filter_nums = 30,40,50
windows_size = 5

[Optimizer]
adam = False
sgd = True
learning_rate = 0.015
weight_decay = 1.0e-8
momentum = 0.0
clip_max_norm_use = False
clip_max_norm = None
use_lr_decay = True
lr_rate_decay = 0.05
min_lrate = 0.000005
max_patience = 1

[Train]
num_threads = 1
use_cuda = True 
epochs = 100
early_max_patience = 10
backward_batch_size = 1
batch_size = 64
dev_batch_size = 64
test_batch_size = 64
log_interval = 100

`

**Comment by bamtercelboo** [comment]
[Model]
use_char = True
try change 'use_char' to False.

My Chinese data format is a Chinese character corresponding to a label, like this.
我 O
们 O
变 O
而 O
以 O
书 O
会 O
友 O
， O
以 O
书 O
结 O
缘 O
， O
把 O
欧 B-LOC
美 B-LOC
、 O
港 B-LOC
台 B-LOC
流 O
行 O
的 O
食 O
品 O
类 O
图 O
谱 O
、 O
画 O
册 O

**Comment by EricAugust** [comment]
For chinese data, only use BILSTM + CRF ?

**Comment by bamtercelboo** [comment]
@EricAugust 
Hi, I haven't tried your Chinese data format. It seems that char can be used, but it is not clear why loss is zero on your side.

**Comment by EricAugust** [comment]
@bamtercelboo 
logit is not null.  There are some problem in crf process.

**Comment by bamtercelboo** [comment]
@EricAugust 
Ok, I will check the CRF section of the code in my spare time, can you sent your sample data to me  for test? or if you see some problems, please tell me in here or email, thanks.

**Comment by EricAugust** [comment]
@bamtercelboo 
Sure, please check your email.

**Comment by bamtercelboo** [comment]
@EricAugust 
Ok, I have received the data, Thanks.

**Comment by bamtercelboo** [comment]
For chinese NER data, readjust the network parameters and optimizer.
